File: ts/management-api-v2/cm_api_v2_patch_taxonomy_group.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = await client.modifyTaxonomy()
  .byTaxonomyId('0be13600-e57c-577d-8108-c8d860330985')
  // .byTaxonomyCodename('personas')
  // .byTaxonomyExternalId('Tax-Group-123')
  .withData(
    [
      {
        op: 'replace',
        property_name: 'name',
        value: 'Categories'
      },
      {
        op: 'replace',
        property_name: 'codename',
        value: 'category'
      },
      {
        op: 'replace',
        reference: {
          codename: 'first_term'
        },
        property_name: 'terms',
        value: [
          {
            name: 'Second-level taxonomy term',
            codename: 'second_term',
            terms: [
              {
                name: 'Third-level taxonomy term',
                terms: []
              }
            ]
          }
        ]
      },
      {
        op: 'remove',
        reference: {
          external_id: 'unused-taxonomy-term'
        }
      },
      {
        op: 'addInto',
        reference: {
          codename: 'second_term'
        },
        value:
        {
          name: 'New taxonomy term',
          external_id: 'my-new-term',
          terms: []
        }
      }
    ]
  )
  .toPromise();

File: net/management-api-v2/cm_api_v2_patch_taxonomy_group.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Configuration;
using Kontent.Ai.Management.Models.Shared;
using Kontent.Ai.Management.Models.TaxonomyGroups;
using Kontent.Ai.Management.Models.TaxonomyGroups.Patch;

// Or register it through DI with services.AddManagementClient()
using var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var identifier = Reference.ById(Guid.Parse("0be13600-e57c-577d-8108-c8d860330985"));
// var identifier = Reference.ByCodename("personas");
// var identifier = Reference.ByExternalId("Tax-Group-123");

var response = (await client.ModifyTaxonomyGroupAsync(identifier,
[
    TaxonomyGroupPatch.ReplaceName(identifier, "Categories"),
    TaxonomyGroupPatch.ReplaceCodename(identifier, "category"),
    TaxonomyGroupPatch.ReplaceTerms(Reference.ByCodename("first_term"),
        new TaxonomyTermCreateModel
        {
            Name = "Second-level taxonomy term",
            Codename = "second_term",
            Terms =
            [
                new TaxonomyTermCreateModel
                {
                    Name = "Third-level taxonomy term",
                }
            ]
        }),
    new TaxonomyGroupRemovePatchModel
    {
        Reference = Reference.ByExternalId("unused-taxonomy-term")
    },
    new TaxonomyGroupAddIntoPatchModel
    {
        Reference = Reference.ByCodename("second_term"),
        Value = new TaxonomyTermCreateModel
        {
            Name = "New taxonomy term",
            ExternalId = "my-new-term",
        }
    },
    new TaxonomyGroupMovePatchModel
    {
        Reference = Reference.ByExternalId("my-new-term"),
        Before = Reference.ByCodename("first_term")
    }
])).EnsureSuccess();
